yearn/v2/registry.py

```diff
@@ -69,6 +70,8 @@ def watch_events(self):
             if not self._done.is_set():
                 self._done.set()
                 logger.info("loaded v2 registry in %.3fs", time.time() - start)
+            if not config.watch_events_forever:
```

Comment:
Maybe this behavior should be defined by a constructor argument and not global config? It would make it easier to reason about.

Reply:
moved it to a constructor arg to Registry, Strategy and Vault